grblas/formatting.py

```diff
@@ -202,18 +204,79 @@ def _get_matrix_dataframe(matrix, max_rows, min_rows, max_columns, *, mask=None)
                 column_offset,
                 mask=mask,
             )
+    if (
+        (mask is None or mask.structure)
+        and df.shape != matrix.shape
+        and min(matrix._nvals, max_rows if matrix._nvals <= max_rows else min_rows)
+        > 2 * df.count().sum()
```

Comment:
Note to self: this condition was chosen so I wouldn't need to rewrite any tests.  "Friendlier" conditions for when to switch to the sparse repr may exist.